### ISSUE TITLE
triple协议如何能够配置随机端口呢？同一台机器部署多个相同服务，端口会重复，java中协议端口可以允许-1端口随机分配

### ISSUE THREAD

**Issue by tiandi0716** [apache/dubbo-go#2382]
![image](https://github.com/apache/dubbo-go/assets/41275334/d047eda7-1ed2-4a8a-91b9-628d9fdb940f)

调整代码

![image](https://github.com/apache/dubbo-go/assets/41275334/9dfc4724-d99a-411c-9daf-dae1f1441c5d)

不要默认值是不是才能通过后面逻辑？，否则后面判断端口逻辑是无效的？

![image](https://github.com/apache/dubbo-go/assets/41275334/8aa3f03d-36c8-4987-9c4d-cd82e7c3d41c)

**Comment by justxuewei** [comment]
Thank you for reporting this. Firstly, the documentation you referenced is from Dubbo, not Dubbo-go. Therefore, the default port is not the same as the Dubbo's. Then, the condition of using ramdon port doesn't have problem, since the port should be specified to `-1`, instead of leaving empty.

**Comment by tiandi0716** [comment]
@justxuewei  Thank you your reply, But use  "-1"  in port field of string type,  the length  size is greater than 0,  if  i keep empty  the field default is "20000",  

![image](https://github.com/apache/dubbo-go/assets/41275334/1caff008-09e6-4c22-a3aa-b9d12429e4a3)

there  you are  set the "default" tag value "20000", please give a simple of how to use random port?

 thank you very much

**Comment by justxuewei** [comment]
> use "-1" in port field of string type, the length size is greater than 0

What is happend while the length size is greater than 0? Does it not work while setting the port to `-1`?

**Comment by tiandi0716** [comment]
Protocol  Init  method
 
https://github.com/apache/dubbo-go/blob/main/config/protocol_config.go
![image](https://github.com/apache/dubbo-go/assets/41275334/81f0d4ec-33e1-4f2b-a69a-0f101f36e40a)


https://github.com/creasty/defaults/blob/master/defaults.go
![image](https://github.com/apache/dubbo-go/assets/41275334/06365f35-cee2-4e56-9633-9a2134af47de)

the code you are parse  "default" tag,  Port  default "20000"
https://github.com/apache/dubbo-go/blob/main/config/protocol_config.go
![image](https://github.com/apache/dubbo-go/assets/41275334/1b50be9e-a3d9-429f-9b14-0b36648e225f)

**my   dubbogo.yaml  config**

```
dubbo:
  application: 
    name: go.micro.api.gateway
  registries:
    nacos:
      protocol: nacos
      address: 192.168.1.236:8848
  protocols:
    triple:
      name: tri
      port:  -1
```

**Comment by justxuewei** [comment]
Thanks for report this issue. I've pushed a patch to fix the issue: https://github.com/apache/dubbo-go/pull/2384. Please let me know if you have any questions.

**Comment by tiandi0716** [comment]
Ok,  thank you， we will let you  know  if  i meet any question in use.

**Comment by project-snail** [comment]
> Thanks for report this issue. I've pushed a patch to fix the issue: #2384. Please let me know if you have any questions.

感觉是判断-1才特殊处理？现在无法解析就随机分配 无法解析结束进程panic感觉比较合理
